config/src/config/mod.rs

```diff
@@ -398,15 +394,33 @@ mod test {
 
     #[test]
     fn verify_all_configs() {
+        // First test well defined configs
         let _ = vec![
-            PathBuf::from("data/configs/overrides/persistent_data.node.config.override.toml"),
-            PathBuf::from("data/configs/single.node.config.toml"),
-            PathBuf::from("../terraform/validator-sets/100/fn/node.config.toml"),
-            PathBuf::from("../terraform/validator-sets/100/val/node.config.toml"),
-            PathBuf::from("../terraform/validator-sets/dev/fn/node.config.toml"),
-            PathBuf::from("../terraform/validator-sets/dev/val/node.config.toml"),
+            "data/configs/single.node.config.toml",
         ]
         .iter()
-        .map(|path| NodeConfig::load(path).expect("NodeConfig"));
+        .map(|path| NodeConfig::load(PathBuf::from(path)).expect(&format!("Error in {}", path)))
```

Comment:
The linter is going to complain about the `.expect(&format!("Error in {}", path)))` part look at using `anyhow::Context` and use `with_context().unwrap()` instead if you need to perform and allocation. Even better, maybe add the context in the load function itself so you can just unwrap here.

Reply:
Indeed it did. I wanted to have a completely safe API in the configs in that there should be no panics just errors and delegate panicing to the main functions etc.

Reply:
Hmm.. there's a lot of different places errors could occur, it would seem we would need to pass back context on each.